### PR TITLE
Switch ApiScan to use Azure CLI auth

### PR DESCRIPTION
**Pull request by weshaggard**
Switch ApiScan to use Azure CLI auth. While I'm not particularly happy with this approach, the APIScan tool/task doesn't support WIF directly currently so we are moving to use the Azure CLI login instead.

### CODE REVIEW

File: eng/pipelines/aggregate-reports.yml [hunk 1]
```diff
@@ -168,6 +168,26 @@ stages:
...
+              addSpnToEnvironment: true
+              inlineScript: |
+                az --version
+                az account show
+                Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($env:servicePrincipalId)"
+                Write-Host "##vso[task.setvariable variable=ARM_TENANT_ID;issecret=true]$($env:tenantId)"
+                Write-Host "##vso[task.setvariable variable=ARM_OIDC_TOKEN;issecret=true]$($env:idToken)"
+
+          - pwsh: |
+                az login --service-principal -u $(ARM_CLIENT_ID) --tenant $(ARM_TENANT_ID) --allow-no-subscriptions --federated-token $(ARM_OIDC_TOKEN)
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:186]: @benbp can you double check this work and let me know if this is too hacky? At the root I need to have a login credential to pass to the apiscan task. The CLI nor PS stay logged in so I'm exporting the token to login and using it here.

File: eng/pipelines/aggregate-reports.yml [hunk 2]
```diff
@@ -168,6 +168,27 @@ stages:
...
+                az account show
+                Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($env:servicePrincipalId)"
+                Write-Host "##vso[task.setvariable variable=ARM_TENANT_ID;issecret=true]$($env:tenantId)"
+                Write-Host "##vso[task.setvariable variable=ARM_OIDC_TOKEN;issecret=true]$($env:idToken)"
+
+          - pwsh: |
+                # Need to re-login with the az login so that it presists and can be used in the APISca task
+                az login --service-principal -u $(ARM_CLIENT_ID) --tenant $(ARM_TENANT_ID) --allow-no-subscriptions --federated-token $(ARM_OIDC_TOKEN)
+                az --version
+                az account show
```
Comment (benbp) [inline eng/pipelines/aggregate-reports.yml:189]:
```suggestion
                az account show -o json
```

File: eng/pipelines/aggregate-reports.yml [hunk 3]
```diff
@@ -179,8 +200,14 @@ stages:
               preserveLogsFolder: true
               verbosityLevel: standard
             env:
-              # azure-sdk-apiscan (81109e5f-0620-423c-a37a-c22fbf8973a7)
-              AzureServicesAuthConnectionString: runAs=App;AppId=81109e5f-0620-423c-a37a-c22fbf8973a7;TenantId=72f988bf-86f1-41af-91ab-2d7cd011db47;AppKey=$(azure-sdk-apiscan-client-secret)
+              AzureServicesAuthConnectionString: RunAs=Developer;DeveloperTool=AzureCli
+
+          - pwsh: |
+                az account show
+                az Logout
```
Comment (benbp) [inline eng/pipelines/aggregate-reports.yml:207]:
```suggestion
                az logout
```

File: eng/pipelines/aggregate-reports.yml (hunk 1 above)
Comment (benbp) [reply eng/pipelines/aggregate-reports.yml:187]: I think this is fine, assuming the task doesn't take longer than the token expiry. I'm pretty much using this same az login approach in other places where to bridge the workload identity support gap.

File: eng/pipelines/aggregate-reports.yml (hunk 2 above)
Comment (weshaggard) [reply eng/pipelines/aggregate-reports.yml:189]: What does `-o json` by us? I just tried locally and it appears to dump the same string to the console.

**Review by benbp — APPROVED** [review]

File: eng/pipelines/aggregate-reports.yml (hunk 2 above)
Comment (benbp) [reply eng/pipelines/aggregate-reports.yml:189]: I think in the pipeline it usually defaults to json so the output is the same, but for example in my terminal it usually defaults to tsv, which omits the subscription id. Specifying json explicitly means we always get the intended output.

File: eng/pipelines/aggregate-reports.yml [hunk 4]
```diff
@@ -179,8 +200,14 @@ stages:
               preserveLogsFolder: true
               verbosityLevel: standard
             env:
-              # azure-sdk-apiscan (81109e5f-0620-423c-a37a-c22fbf8973a7)
-              AzureServicesAuthConnectionString: runAs=App;AppId=81109e5f-0620-423c-a37a-c22fbf8973a7;TenantId=72f988bf-86f1-41af-91ab-2d7cd011db47;AppKey=$(azure-sdk-apiscan-client-secret)
+              AzureServicesAuthConnectionString: RunAs=Developer;DeveloperTool=AzureCli
+
+          - pwsh: |
+                az account show
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:206]:
```suggestion
                az account show -o json
```

File: eng/pipelines/aggregate-reports.yml [hunk 5]
```diff
@@ -168,6 +168,27 @@ stages:
...
+          - task: AzureCLI@2
+            displayName: Azure CLI Login
+            inputs:
+              azureSubscription: azure-sdk-apiscan
+              scriptType: pscore
+              scriptLocation: inlineScript
+              addSpnToEnvironment: true
+              inlineScript: |
+                az --version
+                az account show
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:180]:
```suggestion
                az account show -o json
```

File: eng/pipelines/aggregate-reports.yml [hunk 6]
```diff
@@ -168,6 +168,27 @@ stages:
...
+                az --version
+                az account show -o json
+                Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($env:servicePrincipalId)"
+                Write-Host "##vso[task.setvariable variable=ARM_TENANT_ID;issecret=true]$($env:tenantId)"
+                Write-Host "##vso[task.setvariable variable=ARM_OIDC_TOKEN;issecret=true]$($env:idToken)"
+
+          - pwsh: |
+                # Need to re-login with the az login so that it presists and can be used in the APISca task
+                az login --service-principal -u $(ARM_CLIENT_ID) --tenant $(ARM_TENANT_ID) --allow-no-subscriptions --federated-token $(ARM_OIDC_TOKEN)
+                az --version
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:188]:
```suggestion
```

**Pull request merged**